src/lib/sw_engine/tvgSwRaster.cpp

```diff
@@ -176,6 +176,30 @@ static bool _translucentRectInvAlphaMask(SwSurface* surface, const SwBBox& regio
     return true;
 }
 
+
+static bool _translucentRectColorMask(SwSurface* surface, const SwBBox& region, uint32_t color)
+{
+    auto buffer = surface->buffer + (region.min.y * surface->stride) + region.min.x;
+    auto h = static_cast<uint32_t>(region.max.y - region.min.y);
+    auto w = static_cast<uint32_t>(region.max.x - region.min.x);
+
+    TVGLOG("SW_ENGINE", "Rectangle Color Mask Composition");
+
+    auto cbuffer = surface->compositor->image.data + (region.min.y * surface->stride) + region.min.x; //compositor buffer
+
+    for (uint32_t y = 0; y < h; ++y) {
+        auto dst = &buffer[y * surface->stride];
+        auto cmp = &cbuffer[y * surface->stride];
+        for (uint32_t x = 0; x < w; ++x) {
+            auto grayvalue = (((((*cmp)&0xff)*19) + ((((*cmp)>>8)&0xff)*183) + ((((*cmp)>>16)&0xff)*54)) * surface->blender.alpha(*cmp) + 0xffff) >> 16; //(0.0721*B + 0.7154*G + 0.2125*R) * A
```

Comment:
rgb in cmp are already multiplied by alpha, so now you multiply it for the second time.

I'd intruduce a separate function like _grayScale that would return the value that you called grayvalue.

also coding convention:

- grayvalue -> grayValue
- spaces between numbers